app/test/square_size_test.go

```diff
@@ -73,34 +75,38 @@ func (s *SquareSizeIntegrationTest) TestSquareSizeUpperBound() {
 	}
 
 	tests := []test{
-		{
-			name:             "default",
-			govMaxSquareSize: appconsts.DefaultGovMaxSquareSize,
-			maxBytes:         appconsts.DefaultMaxBytes,
-			expMaxSquareSize: appconsts.DefaultGovMaxSquareSize,
-		},
 		{
 			name:             "max bytes constrains square size",
-			govMaxSquareSize: appconsts.DefaultGovMaxSquareSize,
+			govMaxSquareSize: 64,
 			maxBytes:         appconsts.DefaultMaxBytes,
-			expMaxSquareSize: appconsts.DefaultGovMaxSquareSize,
-		},
-		{
-			name:             "gov square size == hardcoded max",
-			govMaxSquareSize: appconsts.SquareSizeUpperBound,
-			maxBytes:         appconsts.DefaultUpperBoundMaxBytes,
-			expMaxSquareSize: appconsts.SquareSizeUpperBound,
+			expMaxSquareSize: 64,
 		},
 	}
 
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	errCh := make(chan error)
 	go func() {
-		seqs := txsim.NewBlobSequence(txsim.NewRange(100_000, 100_000), txsim.NewRange(1, 1)).Clone(100)
-		opts := txsim.DefaultOptions().WithSeed(rand.Int63()).WithPollTime(time.Second).SuppressLogs()
-		errCh <- txsim.Run(ctx, s.grpcAddr, s.cctx.Keyring, s.enc, opts, seqs...)
+		// this for loop is a hack to restart txsim on failures
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
```

Comment:
hacky, as a flup we should enabled txsim to handle failures gracefully instead of returning after 1

see other comments for converting this test to one or more unit tests